tests/libtest/test_ibmcloud.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf8 -*-
+
+
+import logging
+
+from ocs_ci.ocs.platform_nodes import IBMCloud
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import libtest
+from ocs_ci.ocs import constants, node
+from ocs_ci.ocs.node import get_all_nodes
+from ocs_ci.framework.pytest_customization.marks import ibmcloud_platform_required
+
+from ocs_ci.utility.utils import run_cmd
+
+logger = logging.getLogger(__name__)
+
+
+@libtest
+@ibmcloud_platform_required
+def test_rebootnodes():
+    """
+    Check basic consistency in platform handling.
+    """
+    ibmcloud = IBMCloud()
+    worker_nodes = node.get_nodes(node_type="worker", num_of_nodes=1)
+    ibmcloud.restart_nodes(worker_nodes)
+
+@libtest
+@ibmcloud_platform_required
+def test_attachvolume():
+    """
+    Check basic consistency in platform handling.
+    """
+    ibmcloud = IBMCloud()
+
+    volume = ibmcloud.getVolumeId()
+
+    worker_nodes = node.get_nodes(node_type="worker", num_of_nodes=1)
+    ibmcloud.attach_volume(volume, worker_nodes)
+
+@libtest
+@ibmcloud_platform_required
+def test_detachvolume():
```

Comment:
I think this test should use fixture which will create the volume in it's setup phase. Than you test will use this volume to actually detach it. The finalizer of this fixture should remove the volume. So no leftovers should be there after the test has finished.

Reply:
Yes. I am detaching the volume where we created and attached as part of testing

Reply:
I think you didn't get me.  Please take a look at example I shared in some comment above:
https://stackoverflow.com/questions/44716237/pytest-passing-data-for-cleanup

The tests should be independent on each other and even if you will run one test out of this it should succeed as test itself.  The fixtures makes live easier as it provides (setup) resources for you and you can then write finalizer which will just remove resource from teardown of the fixture. You can use different scopes so if the resources should be shared by whole execution then you can make it session scope - (no test should delete such resource) only finalizer of that fixture.

Or Function/Class scope.

If you didn't get my point we can have a call so I can explain more.